codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/DocumentClientCommandGenerator.java

```diff
@@ -0,0 +1,380 @@
+import software.amazon.smithy.model.knowledge.OperationIndex;
+import software.amazon.smithy.model.shapes.CollectionShape;
+import software.amazon.smithy.model.shapes.MapShape;
+import software.amazon.smithy.model.shapes.MemberShape;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.shapes.StructureShape;
+import software.amazon.smithy.model.traits.IdempotencyTokenTrait;
+import software.amazon.smithy.typescript.codegen.ApplicationProtocol;
+import software.amazon.smithy.typescript.codegen.TypeScriptSettings;
+import software.amazon.smithy.typescript.codegen.TypeScriptWriter;
+
+final class DocumentClientCommandGenerator implements Runnable {
+    static final String COMMAND_PROPERTIES_SECTION = "command_properties";
+    static final String COMMAND_BODY_EXTRA_SECTION = "command_body_extra";
+    static final String COMMAND_CONSTRUCTOR_SECTION = "command_constructor";
+
+    private final Model model;
+    private final OperationShape operation;
+    private final SymbolProvider symbolProvider;
+    private final TypeScriptWriter writer;
+    private final Symbol symbol;
+    private final OperationIndex operationIndex;
+    private final String inputTypeName;
+    private final List<MemberShape> inputMembersWithAttr;
+    private final String outputTypeName;
+    private final List<MemberShape> outputMembersWithAttr;
+
+    DocumentClientCommandGenerator(
+            TypeScriptSettings settings,
+            Model model,
+            OperationShape operation,
+            SymbolProvider symbolProvider,
+            TypeScriptWriter writer
+    ) {
+        this.model = model;
+        settings.getService(model);
+        this.operation = operation;
+        this.symbolProvider = symbolProvider;
+        this.writer = writer;
+
+        symbol = symbolProvider.toSymbol(operation);
+        operationIndex = OperationIndex.of(model);
+        inputTypeName = DocumentClientUtils.getModifiedName(
+            symbol.expectProperty("inputType", Symbol.class).getName()
+        );
+        inputMembersWithAttr = getStructureMembersWithAttr(operationIndex.getInput(operation));
+        outputTypeName = DocumentClientUtils.getModifiedName(
+            symbol.expectProperty("outputType", Symbol.class).getName()
+        );
+        outputMembersWithAttr = getStructureMembersWithAttr(operationIndex.getOutput(operation));
+    }
+
+    @Override
+    public void run() {
+        String serviceName = DocumentClientUtils.CLIENT_NAME;
+        String configType = DocumentClientUtils.CLIENT_CONFIG_NAME;
+
+        // Add required imports.
+        writer.addImport(configType, configType, "./" + serviceName);
+        writer.addImport("Command", "$Command", "@aws-sdk/smithy-client");
+
+        generateInputAndOutputTypes();
+
+        String name = DocumentClientUtils.getModifiedName(symbol.getName());
+        // writer.writeShapeDocs(operation);
```

Comment:
The function `writeShapeDocs` is not visible on TypeScriptWriter https://github.com/awslabs/smithy-typescript/blob/944ac96655b95bf00a171fe933523c78deada209/smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/TypeScriptWriter.java#L216-L226

Since we're mentioning in the documentation that it accepts JavaScript native types, I'll write a custom wrapper for `writeDocs` in `DocumentClientUtils`.

Reply:
Added doc comments in [`fa2e8db` (#2097)](https://github.com/aws/aws-sdk-js-v3/pull/2097/commits/fa2e8db47564dca1a95593fc720c2a3db5ede7be)